server/dex/dex.go

```diff
@@ -390,6 +391,17 @@ func NewDEX(cfg *DexConf) (*DEX, error) {
 	authMgr := auth.NewAuthManager(&authCfg)
 	startSubSys("Auth manager", authMgr)
 
+	markets := make(map[string]*market.Market, len(cfg.Markets))
+	marketUnbookHook := func(lo *order.LimitOrder) bool {
+		name, err := dex.MarketName(lo.BaseAsset, lo.QuoteAsset)
+		if err != nil {
+			log.Error(err)
+			return false
+		}
+
+		return markets[name].Unbook(lo)
+	}
```

Comment:
This unbookHook function is a little bridge created by the DEX manager.  I'm not super thrilled with this, but I also don't  like that the order that is failing swaps should stay on the books.  This little callback is a way for Swapper to communicate up to the DEX manager of the desire to unbook the order so that DEX can make the request to the relevant market.